Lib/ufo2ft/instantiator.py

```diff
@@ -596,40 +689,141 @@ def _generate_instance_info(
+                    "that is not one of the standard values ('Regular', 'Italic', 'Bold', "
+                    "'Bold Italic'). This may cause problems in some applications.",
+                    location,
+                )
+            # convert to lowercase as per UFO spec
+            font.info.styleMapStyleName = instance.styleMapStyleName.lower()
+
+        # multilingual names overriding; specific for OpenType name records on Windows
+        mulitlingual_opentype_font_records = []
+        # list out all language tags that are used in the instance
+        declared_language_tags = set(
+            instance.localisedFamilyName.keys()
+            | instance.localisedStyleName.keys()
+            | instance.localisedStyleMapFamilyName.keys()
+            | instance.localisedStyleMapStyleName.keys()
+        )
+        OT_TYPOGRAPHIC_FAMILY_NAME_ID = 16
+        OT_TYPOGRAPHIC_SUBFAMILY_NAME_ID = 17
+        OT_STYLE_MAP_FAMILY_NAME_ID = 1
+        OT_STYLE_MAP_STYLE_NAME_ID = 2
+        for language_tag in declared_language_tags:
+            if font.info.openTypeNameRecords is None:
+                font.info.openTypeNameRecords = []
+            # currently only make records for Windows language tag (platform 3, platform encoding 1)
+            if language_tag in instance.localisedFamilyName:
+                # assume is Preferred Family Name (ID=16)
+                mulitlingual_opentype_font_records.append(
+                    _make_ufo_name_record(
+                        OT_TYPOGRAPHIC_FAMILY_NAME_ID,
+                        instance.localisedFamilyName[language_tag],
+                        language_tag,
+                    )
+                )
+            if language_tag in instance.localisedStyleName:
+                # assume is Preferred Subfamily Name (ID=17)
+                mulitlingual_opentype_font_records.append(
+                    _make_ufo_name_record(
+                        OT_TYPOGRAPHIC_SUBFAMILY_NAME_ID,
+                        instance.localisedStyleName[language_tag],
+                        language_tag,
+                    )
+                )
+            if language_tag in instance.localisedStyleMapFamilyName:
+                # Style Map Family Name (ID=1)
+                mulitlingual_opentype_font_records.append(
+                    _make_ufo_name_record(
+                        OT_STYLE_MAP_FAMILY_NAME_ID,
+                        instance.localisedStyleMapFamilyName[language_tag],
+                        language_tag,
+                    )
+                )
+            elif language_tag in instance.localisedFamilyName:
+                # Style Map Family Name (ID=1)
+                # autofallback to ID=16 + styleName ID=17 (but omit ID=17 if it's standard style)
+                full_family_name = instance.localisedFamilyName[language_tag]
+                if language_tag in instance.localisedStyleName:
+                    # if the styleMapStyleName is set, use it
```

Comment:
this inline comment doesn't make sense to me, it's referring to `styleMapStyleName` but you're handling `localisedStyleName` here..